### PR TITLE
Added several bot animations

### PR DESCRIPTION
**Pull request by EnigmaticDragon**
Bots now have proper animations for melee attacks and dashing. Also they hold the rifle correctly.

### CODE REVIEW

File: Assets/Scripts/Bots/AIMBotController.cs [hunk 1]
```diff
@@ -124,21 +125,23 @@ private void Awake()
             }
 
             navMeshAgent = GetComponent<NavMeshAgent>();
-            if (navMeshAgent == null)
+            if (meleeWeapon == null)
```
Comment (lee17017) [inline Assets/Scripts/Bots/AIMBotController.cs]: It does not look like this change to meleeWeapon is intended

File: Assets/MeleeWeaponBehaviour.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
+﻿using AIM.Bots;
```
Comment (nfetissow) [inline Assets/MeleeWeaponBehaviour.cs]: Please move the file to the belonging folder inside the Assets/Scripts

File: Assets/Scripts/Bots/AIMBotController.cs (hunk 1 above)
Comment (nfetissow) [reply Assets/Scripts/Bots/AIMBotController.cs]: yep, good catch

File: Assets/Scripts/Utilities/CoroutineCancellationToken.cs [hunk 3]
```diff
@@ -21,5 +21,8 @@ public static IEnumerator TimedCancel(this CoroutineCancellationToken token, flo
             yield return new UnityEngine.WaitForSeconds(time);
             token.CancellationRequested = true;
         }
+
+        // for lazy null check
+        public static void Cancel(this CoroutineCancellationToken token) => token.CancellationRequested = true;
```
Comment (nfetissow) [inline Assets/Scripts/Utilities/CoroutineCancellationToken.cs]: I like this change :)

File: Assets/Scripts/Bots/AIMBotController.cs (hunk 1 above)
Comment (EnigmaticDragon) [reply Assets/Scripts/Bots/AIMBotController.cs]: Oops 🤔Nope, wasn't intended.

**Pull request merged**